mysql/mysqldump.go

```diff
@@ -186,24 +186,29 @@ func processCreateTable(conv *internal.Conv, stmt *ast.CreateTableStmt) {
 	var colNames []string
 	colDef := make(map[string]schema.Column)
 	var keys []schema.Key
+	var fkeys []schema.ForeignKey
 	for _, element := range stmt.Cols {
-		colname, col, isPk, err := processColumn(conv, tableName, element)
+		colname, col, constraint, err := processColumn(conv, tableName, element)
 		if err != nil {
 			logStmtError(conv, stmt, err)
 			return
 		}
 		colNames = append(colNames, colname)
 		colDef[colname] = col
-		if isPk {
+		if constraint.isPk {
 			keys = append(keys, schema.Key{Column: colname})
 		}
+		if constraint.fk.Columns != nil {
```

Comment:
Why is this needed?
Can we construct a foreign key constraint with zero-length cols?
I don't think MySQL allows that.